python/adbc_driver_manager/adbc_driver_manager/_lib.pyi

```diff
@@ -0,0 +1,130 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any, ClassVar, Dict, List, Optional, Tuple, Union
+
+from typing import overload
+import enum
+import typing
+
+INGEST_OPTION_MODE: str
+INGEST_OPTION_MODE_APPEND: str
+INGEST_OPTION_MODE_CREATE: str
+INGEST_OPTION_TARGET_TABLE: str
+
+class AdbcConnection(_AdbcHandle):
+    def __init__(self, *args, **kwargs) -> None: ...
+    def close(self, *args, **kwargs) -> Any: ...
+    def commit(self, *args, **kwargs) -> Any: ...
+    def get_info(self, *args, **kwargs) -> Any: ...
+    def get_objects(self, *args, **kwargs) -> Any: ...
+    def get_table_schema(self, *args, **kwargs) -> Any: ...
+    def get_table_types(self, *args, **kwargs) -> Any: ...
+    def read_partition(self, *args, **kwargs) -> Any: ...
+    def rollback(self, *args, **kwargs) -> Any: ...
+    def set_autocommit(self, *args, **kwargs) -> Any: ...
+    def set_options(self, *args, **kwargs) -> Any: ...
+
+class AdbcDatabase(_AdbcHandle):
+    def __init__(self, **kwargs: Dict[str, str]) -> None: ...
+    def close(self, *args, **kwargs) -> Any: ...
+    def set_options(self, *args, **kwargs) -> Any: ...
+
+class AdbcInfoCode(enum.IntEnum):
+    DRIVER_ARROW_VERSION: ClassVar[AdbcInfoCode] = ...
```

Comment:
Were the `ClassVar` annotations here auto-generated? I probably have out of date knowledge on annotations but haven't seen that pattern before

Reply:
Yeah, with mypy's stubgen. I already hand-edited it somewhat to clean things up but it looks like it doesn't get any function signatures _at all_ so it'll require some more hand editing.

Reply:
Makes sense. I think this is a good starting point as is - can always be refined later